applications/ConvectionDiffusionApplication/custom_strategies/strategies/explicit_runge_kutta_4_eulerian_convdiff_strategy.h

```diff
@@ -0,0 +1,385 @@
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /**
+     * @brief Initialize the Runge-Kutta substep.
+     * @details Calculate the orthogonal subscale projections if required.
+     */
+    virtual void InitializeRungeKuttaIntermediateSubStep() override
+    {
+        KRATOS_TRY;
+
+        BaseType::InitializeRungeKuttaIntermediateSubStep();
+
+        auto& r_model_part = BaseType::GetModelPart();
+        auto& r_process_info = r_model_part.GetProcessInfo();
+        // set interpolation parameter TIME_INTEGRATION_THETA
+        if (r_process_info.GetValue(RUNGE_KUTTA_STEP) == 1) {
+            r_process_info.GetValue(TIME_INTEGRATION_THETA) = 1.0;
+        }
+        else if (r_process_info.GetValue(RUNGE_KUTTA_STEP) == 2 || r_process_info.GetValue(RUNGE_KUTTA_STEP) == 3) {
+            r_process_info.GetValue(TIME_INTEGRATION_THETA) = 0.5;
+        }
+        else {
+            KRATOS_ERROR << "RUNGE-KUTTA step not correctly set.";
+        }
+        // execute OSS step, if needed
+        if (r_process_info[OSS_SWITCH] == 1) {
+            ExecuteOSSStep();
+        }
+
+        KRATOS_CATCH("");
+    };
+
+    /**
+     * @brief Initialize the Runge-Kutta substep.
+     * @details Calculate the orthogonal subscale projections if required.
+     */
+    virtual void InitializeRungeKuttaLastSubStep() override
+    {
+        KRATOS_TRY;
+
+        BaseType::InitializeRungeKuttaLastSubStep();
+        auto& r_model_part = BaseType::GetModelPart();
+        auto& r_process_info = r_model_part.GetProcessInfo();
+        // set interpolation parameter TIME_INTEGRATION_THETA
+        if (r_process_info.GetValue(RUNGE_KUTTA_STEP) == 4) {
+            r_process_info.GetValue(TIME_INTEGRATION_THETA) = 0.0;
+        }
+        else {
+            KRATOS_ERROR << "RUNGE-KUTTA step not correctly set.";
+        }
+        // execute OSS step, if needed
+        if (r_process_info[OSS_SWITCH] == 1) {
+            ExecuteOSSStep();
+        }
+
+        KRATOS_CATCH("");
+    };
+
+    /**
+     * @brief Finalize the Runge Kutta explicit solver step.
+     * @details Calculate the orthogonal subscale projections if required.
+     */
+    virtual void FinalizeSolutionStep() override
+    {
+        KRATOS_TRY;
+
+        auto& r_model_part = BaseType::GetModelPart();
+        auto& r_process_info = r_model_part.GetProcessInfo();
+        // set interpolation parameter TIME_INTEGRATION_THETA
+        r_process_info.GetValue(TIME_INTEGRATION_THETA) = 0.0;
+        // execute OSS step, if needed
+        if (r_process_info[OSS_SWITCH] == 1) {
+            ExecuteOSSStep();
+        }
+        BaseType::FinalizeSolutionStep();
+
+        KRATOS_CATCH("");
+    };
+
+    /**
+     * @brief Execute the OSS step.
+     */
+    virtual void ExecuteOSSStep()
```

Comment:
I'd call this something like `CalculateOSSNodalProjections` or something of the sort (this is not a step in a RK sense).

Reply:
This method can be `private`.

Reply:
Done.